code/modules/power/catalytic_engine.dm

```diff
@@ -0,0 +1,519 @@
+#define GEN_ANODE 1
+#define GEN_CATHODE 2
+
+#define UNIT_OPEN 0
+#define UNIT_INACTIVE 1
+#define UNIT_ACTIVE 2
+
+/obj/decal/fakeobjects/catalytic_doodad
+	name = "catalytic generator pipe"
+	desc = "Pipe section allowing a catalytic rod to contact outside fluid for catalysis."
+	icon = 'icons/obj/machines/catalysis.dmi'
+	icon_state = "doodad"
+	anchored = 1
+
+/obj/machinery/power/catalytic_generator
+	name = "catalytic generator core"
+	desc = "Harnesses catalysts' reactions with a large body of appropriate fluid to generate electricity."
+	icon = 'icons/obj/machines/catalysis.dmi'
+	icon_state = "core"
+	anchored = 1
+	density = 1
+
+	var/obj/machinery/catalytic_rod_unit/anode_unit
+	var/obj/machinery/catalytic_rod_unit/cathode_unit
+
+	var/gen_rate = 0
+	var/defer_updateicon = FALSE
+	var/output_tier = 0
+	var/datum/light/light
+
+	var/sound_grump = 'sound/machines/buzz-two.ogg'
+
+	New()
+		..()
+
+		light = new /datum/light/point
+		light.attach(src)
+
+		SPAWN(0.5 SECONDS)
+			src.anode_unit = locate(/obj/machinery/catalytic_rod_unit) in get_step(src,WEST)
+			src.cathode_unit = locate(/obj/machinery/catalytic_rod_unit) in get_step(src,EAST)
+			if(!src.anode_unit || !src.cathode_unit)
+				src.status |= BROKEN
+
+			UpdateIcon()
+
+	disposing()
+		src.anode_unit = null
+		src.cathode_unit = null
+		..()
+
+	attack_hand(mob/user,var/bot_operated)
+		if(!src.anode_unit || !src.cathode_unit)
+			boutput(user,"\The [src] doesn't seem to be operable.")
+			return
+
+		if(src.anode_unit.toggling || src.cathode_unit.toggling)
+			boutput(user,"\The [src] doesn't respond to your input.")
+			return
+
+		if(src.anode_unit.mode == UNIT_OPEN && src.cathode_unit.mode == UNIT_OPEN)
+			src.anode_unit.update_mode(UNIT_INACTIVE)
+			src.cathode_unit.update_mode(UNIT_INACTIVE)
+			boutput(user,"You [bot_operated ? "send a toggle signal to" : "press the front button on"] [src]. The catalytic rod units close for operation.")
+			return
+		else
+			src.anode_unit.update_mode(UNIT_OPEN)
+			src.cathode_unit.update_mode(UNIT_OPEN)
+			boutput(user,"You [bot_operated ? "send a toggle signal to" : "press the front button on"] [src]. The catalytic rod units open up.")
+			return
+
+	attack_ai(mob/user)
+		return attack_hand(user,TRUE)
+
+	update_icon()
+		src.defer_updateicon = TRUE
+		if(status & NOPOWER)
+			UpdateOverlays(null, "power")
+		else if(status & BROKEN)
+			UpdateOverlays(image('icons/obj/power.dmi', "teg-err"), "power")
+		else
+			if(output_tier != 0)
+				UpdateOverlays(image('icons/obj/power.dmi', "teg-op[output_tier]"), "power")
+			else
+				UpdateOverlays(null, "power")
+
+		switch (output_tier)
+			if(0)
+				light.disable()
+			if(1 to 12)
+				light.set_color(1, 1, 1)
+				light.set_brightness(0.3)
+			if(13 to INFINITY)
+				light.set_color(1, 1, 1)
+				light.set_brightness(0.6)
+				light.enable()
+
+		. = GetOverlayImage("mask")
+		if(.)
+			UpdateOverlays(.,"mask")
```

Comment:
From what I recall (been a bit since I messed with that), the purpose is to ensure any mask overlays applied by other sources will go over the top of the power indicators, but the one reference I found to that (flock ping) appears to now create a dummy object instead so I'm going to assume it's no longer necessary.